flow/src/lib.rs

```diff
@@ -0,0 +1,975 @@
+	id: Hash,
+	// hash of the overarching body from module-control
+	org: Hash,
+	// name
+	name: Vec<u8>,
+
+	// controller account -> must match body controller
+	// during campaing runtime controller change is not allowed
+	// needs to be revised to avoid campaign attack by starting
+	// a campagin when dao wants to remove controller for reasons
+	owner: AccountId,
+
+	// TODO: THIS NEEDS TO BE GAMEDAO COUNCIL
+	/// admin account of the campaign (operator)
+	admin: AccountId,
+
+	// TODO: assets => GAME
+	/// campaign owners deposit
+	deposit: Balance,
+
+	// TODO: /// campaign start block
+	// start: BlockNumber,
+	/// block until campaign has to reach cap
+	expiry: BlockNumber,
+	/// minimum amount of token to become a successful campaign
+	cap: Balance,
+
+	/// protocol after successful raise
+	protocol: FlowProtocol,
+	/// governance after successful raise
+	governance: FlowGovernance,
+
+	/// content storage
+	cid: Vec<u8>,
+
+	// TODO: prepare for launchpad functionality
+	// token cap
+	// token_cap: u64,
+	// token_price
+	// token_price: u64,
+	// /// token symbol
+	token_symbol: Vec<u8>,
+	// /// token name
+	token_name: Vec<u8>,
+
+	/// creation timestamp
+	created: Timestamp,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type WeightInfo: frame_system::weights::WeightInfo;
+		type Event: From<Event<Self>>
+			+ IsType<<Self as frame_system::Config>::Event>
+			+ Into<<Self as frame_system::Config>::Event>;
+
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = CurrencyId, Balance = Balance>
+			+ MultiReservableCurrency<Self::AccountId>;
+		type UnixTime: UnixTime;
+		type Randomness: Randomness<Self::Hash, Self::BlockNumber>;
+		type Control: ControlPalletStorage<Self::AccountId, Self::Hash>;
+
+		/// The origin that is allowed to make judgements.
+		type GameDAOAdminOrigin: EnsureOrigin<Self::Origin>;
+		type GameDAOTreasury: Get<Self::AccountId>;
+
+		#[pallet::constant]
+		type MinLength: Get<u32>;
+		#[pallet::constant]
+		type MaxLength: Get<u32>;
+
+		#[pallet::constant]
+		type MaxCampaignsPerAddress: Get<u32>;
+		#[pallet::constant]
+		type MaxCampaignsPerBlock: Get<u32>;
+		#[pallet::constant]
+		type MaxContributionsPerBlock: Get<u32>;
+
+		#[pallet::constant]
+		type MinDuration: Get<Self::BlockNumber>;
+		#[pallet::constant]
+		type MaxDuration: Get<Self::BlockNumber>;
+		#[pallet::constant]
+		type MinCreatorDeposit: Get<Balance>;
+		#[pallet::constant]
+		type MinContribution: Get<Balance>;
+
+		#[pallet::constant]
+		type FundingCurrencyId: Get<CurrencyId>;
+
+		// TODO: collect fees for treasury
+		#[pallet::constant]
+		type CampaignFee: Get<Permill>;
+	}
+
+	/// Campaign
+	#[pallet::storage]
+	#[pallet::getter(fn campaign_by_id)]
```

Comment:
Can you also add a ToDo somewhere in the file to update storage objects and drop getters as we discussed earlier?